### ISSUE TITLE
Code Error：NBA数据集部分关于训练数据——速度的计算是否有问题

### ISSUE THREAD

**Issue by SamuelYH-1** [DSL-Lab/MoFlow#4]
![Image](https://github.com/user-attachments/assets/919ffa7e-e515-4502-8afb-556ec49691fb)
图中关于帧的维度在第三维，但是在计算速度指标时，却用的是第二维度，这样计算是否有问题，是否需要改正为： past_traj_vel = torch.cat((past_traj_rel[:, :, 1:] - past_traj_rel[:, :, :-1], torch.zeros_like(past_traj_rel[:, :, -1:])), dim=2)

**Comment by SamuelYH-1** [comment]
> 嗨，好 — 感谢您指出这一点！你说得完全正确。这些是从头开始重新训练所有内容后的 NBA 结果。它们与我们在本文中报告的结果相当，在某些情况下甚至更好。
> 
> ```
> 2025-06-26 02:29:48,419   INFO  25499/25500, running inference on 137500 agents (trajectories)
> 2025-06-26 02:29:48,426   INFO  --ADE_min(1.0s): 0.1773706	--FDE_min(1.0s): 0.2464860
> 2025-06-26 02:29:48,427   INFO  --ADE_min(2.0s): 0.3353395	--FDE_min(2.0s): 0.4638037
> 2025-06-26 02:29:48,427   INFO  --ADE_min(3.0s): 0.5097981	--FDE_min(3.0s): 0.6729834
> 2025-06-26 02:29:48,427   INFO  --ADE_min(4.0s): 0.7000840	--FDE_min(4.0s): 0.8805853
> 2025-06-26 02:29:48,427   INFO  --ADE_avg(1.0s): 0.4218505	--FDE_avg(1.0s): 0.8118399
> 2025-06-26 02:29:48,427   INFO  --ADE_avg(2.0s): 1.0244835	--FDE_avg(2.0s): 2.2071208
> 2025-06-26 02:29:48,427   INFO  --ADE_avg(3.0s): 1.6982480	--FDE_avg(3.0s): 3.5642815
> 2025-06-26 02:29:48,427   INFO  --ADE_avg(4.0s): 2.3172486	--FDE_avg(4.0s): 4.5199056
> ```
> 
> 如果我们使用**前面的** MoFlow 预训练检查点，则推理结果如下。
> 
> ```
> 2025-06-26 02:10:39,329   INFO  --ADE_min(1.0s): 0.1968770      --FDE_min(1.0s): 0.2687883
> 2025-06-26 02:10:39,329   INFO  --ADE_min(2.0s): 0.3603085      --FDE_min(2.0s): 0.4876306
> 2025-06-26 02:10:39,329   INFO  --ADE_min(3.0s): 0.5374129      --FDE_min(3.0s): 0.6932281
> 2025-06-26 02:10:39,329   INFO  --ADE_min(4.0s): 0.7262023      --FDE_min(4.0s): 0.8874637
> 2025-06-26 02:10:39,329   INFO  --ADE_avg(1.0s): 0.4809668      --FDE_avg(1.0s): 0.9042453
> 2025-06-26 02:10:39,329   INFO  --ADE_avg(2.0s): 1.1180333      --FDE_avg(2.0s): 2.3539909
> 2025-06-26 02:10:39,329   INFO  --ADE_avg(3.0s): 1.8156255      --FDE_avg(3.0s): 3.7391878
> 2025-06-26 02:10:39,329   INFO  --ADE_avg(4.0s): 2.4520414      --FDE_avg(4.0s): 4.7134229
> ```
> 
> 我将很快更新这些新的检查点。
Got it - really appreciate your response! I'm also glad I could help.
收到，非常感谢你的回复，也很开心自己能够帮到你。
